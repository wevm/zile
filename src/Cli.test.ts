import { spawn } from 'node:child_process'
import { setupRepos, tree } from '../test/utils.js'

const repos = () =>
  setupRepos().then(({ repos }) => repos.filter(({ relative }) => !relative.includes('invalid-')))

describe.each(await repos())('$relative: Cli.run', ({ cwd }) => {
  test('default', async () => {
    if (cwd.includes('error-')) return

    const result = spawn(
      'bun',
      ['./Cli.ts', '--cwd', cwd, '--includes', '**,!**/error-*,!**/node_modules/**'],
      { cwd: import.meta.dirname },
    )

    const promise = Promise.withResolvers<void>()
    result.on('close', (code) =>
      code === 0
        ? promise.resolve()
        : promise.reject(new Error(`Child process exited with code ${code}`)),
    )
    await promise.promise

    expect(await tree(cwd)).toMatchSnapshot('output tree')
  })
})
