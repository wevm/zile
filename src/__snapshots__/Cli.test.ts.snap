// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`'/basic': Cli.run > default > output tree 1`] = `
"
├── dist
│   ├── foo.d.ts
│   ├── foo.d.ts.map
│   ├── foo.js
│   ├── foo.js.map
│   ├── index.d.ts
│   ├── index.d.ts.map
│   ├── index.js
│   └── index.js.map
├── foo.ts
├── index.ts
├── package.json
└── tsconfig.json
"
`;

exports[`'/basic-object-exports': Cli.run > default > output tree 1`] = `
"
├── dist
│   ├── foo.d.ts
│   ├── foo.d.ts.map
│   ├── foo.js
│   ├── foo.js.map
│   ├── index.d.ts
│   ├── index.d.ts.map
│   ├── index.js
│   └── index.js.map
├── foo.ts
├── index.ts
├── package.json
└── tsconfig.json
"
`;

exports[`'/single-entrypoint': Cli.run > default > output tree 1`] = `
"
├── dist
│   ├── foo.d.ts
│   ├── foo.d.ts.map
│   ├── foo.js
│   ├── foo.js.map
│   ├── index.d.ts
│   ├── index.d.ts.map
│   ├── index.js
│   └── index.js.map
├── foo.ts
├── index.ts
├── package.json
└── tsconfig.json
"
`;
